src/objectcontroller.cpp

```diff
@@ -1478,6 +1484,303 @@ void DateObject::getDate(int *day, int *month, int *year)
         *year = 1900;
 }
 
+DateTimeObjectValue::DateTimeObjectValue(const std::string& value) : 
+  day_m(-1), month_m(-1), year_m(-1), hour_m(-1), min_m(-1), sec_m(-1)
+{
+    if (value == "now")
+        return;
+    std::istringstream val(value);
+    char s1, s2, s3, s4, s5;
+    val >> year_m >> s1 >> month_m >> s2 >> day_m >> s3 >> hour_m >> s4 >> min_m >> s5 >> sec_m;
+    wday_m = 0;
+    year_m -= 1900;
+    if ( val.fail() ||
+         val.peek() != std::char_traits<char>::eof() || // workaround for wrong val.eof() flag in uClibc++
+         s1 != '-' || s2 != '-' || s3 != 'T' || s4 != ':' || s5 != ':' ||
+         year_m < 0 || year_m > 255 || month_m < 1 || month_m > 12 || day_m < 1 || day_m > 31 ||
+	 hour_m < 0 || hour_m > 23 || min_m < 0 || min_m > 59 || sec_m < 0 || sec_m > 59 )
```

Comment:
As you mentioned in a comment, `hour_m == 24` is supported, as long as minutes and seconds are zero. I think we must support that exact combination as well.
Moreover, the standard prescribes that receivers should **ignore** telegrams that are not valid:
![image](https://user-images.githubusercontent.com/7327649/117359035-29e42d80-aeb7-11eb-8a81-d891532a695d.png)
These checks should be done in `::doWrite` I believe?